internal/provider/data_source_auth0_client.go

```diff
@@ -10,16 +10,22 @@ import (
 )
 
 func newDataClient() *schema.Resource {
-	return &schema.Resource{
+	clientDataSource := &schema.Resource{
 		ReadContext: readDataClient,
 		Schema:      newClientSchema(),
+		Description: "Data source to retrieve a specific Auth0 Application client by 'client_id' or 'name'.",
 	}
+
+	addOptionalFieldsToSchema(clientDataSource.Schema, "name", "client_id")
+	clientDataSource.Schema["name"].Description = "The name of the client. If not provided, `client_id` must be set."
+	clientDataSource.Schema["client_id"].Description = "The ID of the client. If not provided, `name` must be set."
+
+	return clientDataSource
 }
 
 func newClientSchema() map[string]*schema.Schema {
-	clientSchema := datasourceSchemaFromResourceSchema(newClient().Schema)
+	clientSchema := dataSourceSchemaFromResourceSchema(newClient().Schema)
 	delete(clientSchema, "client_secret_rotation_trigger")
-	addOptionalFieldsToSchema(clientSchema, "name", "client_id")
```

Comment:
We removed this from here because this func is shared between the client and global client data sources and they have slight differences. 

It is now within the data source constructor https://github.com/auth0/terraform-provider-auth0/pull/268/files#diff-26ee78c4b7e0ff860eb24dc2b54e2641436eb4d123cbf1e9573bb77c7ad8359cR19.